File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/callback/ParametersAndReturnsTestDouble.jl
println("getDouble")
if isdefined(Main, :getDouble) && !isnothing(getDouble)
	error("getDouble already defined!")
end
getDouble = callback("getDouble")
println(typeof(getDouble))
println(getDouble)
if typeof(getDouble) != Float64
	error("getDouble not Float64!")
end
callback("setDouble",getDouble)

println("getDoubleVector")
if isdefined(Main, :getDoubleVector) && !isnothing(getDoubleVector)
	error("getDoubleVector already defined!")
end
getDoubleVector = callback("getDoubleVector")
println(eltype(getDoubleVector))
println(getDoubleVector)
if eltype(getDoubleVector) != Float64
	error("getDoubleVector not Float64!")
end
callback("setDoubleVector",getDoubleVector)

println("getDoubleVectorAsList")
if isdefined(Main, :getDoubleVectorAsList) && !isnothing(getDoubleVectorAsList)
	error("getDoubleVectorAsList already defined!")
end
getDoubleVectorAsList = callback("getDoubleVectorAsList")
println(eltype(getDoubleVectorAsList))
println(getDoubleVectorAsList)
if eltype(getDoubleVectorAsList) != Float64
	error("getDoubleVectorAsList not Float64!")
end
callback("setDoubleVectorAsList",getDoubleVectorAsList)

println("getDoubleMatrix")
if isdefined(Main, :getDoubleMatrix) && !isnothing(getDoubleMatrix)
	error("getDoubleMatrix already defined!")
end
getDoubleMatrix = callback("getDoubleMatrix")
println(eltype(getDoubleMatrix))
println(getDoubleMatrix)
if eltype(getDoubleMatrix) != Float64
	error("getDoubleMatrix not Float64!")
end
callback("setDoubleMatrix",getDoubleMatrix)

println("getDoubleMatrixAsList")
if isdefined(Main, :getDoubleMatrixAsList) && !isnothing(getDoubleMatrixAsList)
	error("getDoubleMatrixAsList already defined!")
end
getDoubleMatrixAsList = callback("getDoubleMatrixAsList")
println(eltype(getDoubleMatrixAsList))
println(getDoubleMatrixAsList)
if eltype(getDoubleMatrixAsList) != Float64
	error("getDoubleMatrixAsList not Float64!")
end
callback("setDoubleMatrixAsList",getDoubleMatrixAsList)

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/callback/ParametersAndReturnsTestDouble.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getDouble")
if(getDouble ~= nil) then
	error("getDouble already defined!")
end
getDouble = callback("getDouble")
getDoubleType = type(getDouble)
print(getDoubleType)
print(getDouble)
if(getDoubleType ~= "number") then
	error("getDouble not Double!")
end
callback("setDouble", getDouble)

print("getDoubleVector")
if(getDoubleVector ~= nil) then
	error("getDoubleVector already defined!")
end
getDoubleVector = callback("getDoubleVector")
getDoubleVectorType = type(getDoubleVector[1])
print(getDoubleVectorType)
print(arraysToString(getDoubleVector))
if(getDoubleVectorType ~= "number") then
	error("getDoubleVector not Double!")
end
callback("setDoubleVector", getDoubleVector)

print("getDoubleVectorAsList")
if(getDoubleVectorAsList ~= nil) then
	error("getDoubleVectorAsList already defined!")
end
getDoubleVectorAsList = callback("getDoubleVectorAsList")
getDoubleVectorAsListType = type(getDoubleVectorAsList[1])
print(getDoubleVectorAsListType)
print(arraysToString(getDoubleVectorAsList))
if(getDoubleVectorAsListType ~= "number") then
	error("getDoubleVectorAsList not Double!")
end
callback("setDoubleVectorAsList", getDoubleVectorAsList)

print("getDoubleMatrix")
if(getDoubleMatrix ~= nil) then
	error("getDoubleMatrix already defined!")
end
getDoubleMatrix = callback("getDoubleMatrix")
getDoubleMatrixType = type(getDoubleMatrix[1][1])
print(getDoubleMatrixType)
print(arraysToString(getDoubleMatrix))
if(getDoubleMatrixType ~= "number") then
	error("getDoubleMatrix not Double!")
end
callback("setDoubleMatrix", getDoubleMatrix)

print("getDoubleMatrixAsList")
if(getDoubleMatrixAsList ~= nil) then
	error("getDoubleMatrixAsList already defined!")
end
getDoubleMatrixAsList = callback("getDoubleMatrixAsList")
getDoubleMatrixAsListType = type(getDoubleMatrixAsList[1][1])
print(getDoubleMatrixAsListType)
print(arraysToString(getDoubleMatrixAsList))
if(getDoubleMatrixAsListType ~= "number") then
	error("getDoubleMatrixAsList not Double!")
end
callback("setDoubleMatrixAsList", getDoubleMatrixAsList)
